src/libraries/System.Text.Json/tests/System.Text.Json.Tests/SourceGenRegressionTests/Net90/Net90GeneratedContext.PropertyNames.g.cs

```diff
@@ -0,0 +1,24 @@
+﻿// <auto-generated/>
+
+#nullable enable annotations
+#nullable disable warnings
+
+// Suppress warnings about [Obsolete] member usage in generated code.
+#pragma warning disable CS0612, CS0618
+
+namespace System.Text.Json.Tests.SourceGenRegressionTests.Net90
+{
+    public partial class Net90GeneratedContext
+    {
+        private static readonly global::System.Text.Json.JsonEncodedText PropName_High = global::System.Text.Json.JsonEncodedText.Encode("High");
```

Comment:
Unrelated from this PR, but just curious: how come the generator is not emitting an UTF8 string literal here instead of a string literal? Wouldn't the UTF8 pick up the more efficient `Encode` overload that would not need to transcode to UTF8 with a temporary buffer, and it could just validate it and then copy the buffer directly, saving a bit of time?

Assuming the reason might be that the generator needs to also target downlevel TFMs which might not have C# 11, we could just check the language version and generate the better pattern when that is available. Or is there some other reason why this overload is being preferred specifically? 🙂

Reply:
> Assuming the reason might be that the generator needs to also target downlevel TFMs which might not have C# 11

Conditionally targeting language features is introducing unnecessary complexity in the generator IMO. The current minimum supported version is C# 9, we can consider removing these once we move it past 11.